lib/super_digit.rb

```diff
@@ -1,15 +1,28 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n) where n is a function of the input length
+# Space Complexity - O(nlog(n)), adding n frames and creating new array each frame because of .digits (each array is of size n where n is a progressively smaller number)
 def super_digit(n)
-   
+  return n if n < 10
+  return super_digit(n.digits.sum)
+  # return super_digit(sum_all_digits(n))
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(nlog(n))
+# since we are only using k to modify initial value of n, time/space complexity is same as super_digit
 def refined_super_digit(n, k)
-    
+    super_digit(n * k)
```

Comment:
You can make this a bit simpler and reduce the time complexity by doing some dynamic programming

```suggestion
  sum = super_digit(n)
  sum *= k
  return super_digit(sum)
```